slice-core/src/main/java/com/cognifide/slice/core/internal/monitoring/InjectorStatisticsRepositoryImpl.java

```diff
@@ -0,0 +1,82 @@
+/*-
+ * #%L
+ * Slice - Core
+ * %%
+ * Copyright (C) 2012 Cognifide Limited
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.slice.core.internal.monitoring;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Queue;
+
+import com.cognifide.slice.api.execution.ExecutionContext;
+import com.cognifide.slice.api.injector.InjectorNameProvider;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+@Singleton
+public class InjectorStatisticsRepositoryImpl implements InjectorStatisticsRepository {
+
+	private final Map<String, ModelUsageData> injectionsStatistics = new HashMap<String, ModelUsageData>();
+
+	private final String injectorName;

```

Comment:
Do we actually need an injectorName in this object?
We are already in the context of given injector and we can always find the name using InjectorHierarchy.getRegisteredName()...
I we could get rid of this injector name that we could remove quite a lot of unnecessary code as a result.